packages/shared/src/components/Feed.tsx

```diff
@@ -263,6 +264,8 @@ export default function Feed<T>({
     }
   }, [selectedPost]);
 
+  useStreakMilestone();
```

Comment:
Should probably be at MainLayout? This is so when the user is visiting a post page, and clicked read article, it would count as view - the milestone popup would be possible to be triggered.

Reply:
To show the modals, the hook needs to be actually executed, which is done on render. If we put it on `MainLayout`, how do we trigger the render? AFAIK we rarely render the whole layout.

Reply:
Asking product if it's OK to show these modals on feed pages only: https://dailydotdev.slack.com/archives/C02E2C3C13R/p1708421020306569?thread_ts=1708099375.192589&cid=C02E2C3C13R

Reply:
Not sure I follow your comment, we have many of such hooks in the past on mainLayout afaik :) 
(the modals that show on action)

@sshanzel will know a lot about this.
Don't see why we would have to limit ourselves.

Reply:
Yeah, also, there can be a case when the user visits a post page link -> clicks read article -> reads the article in a different tab -> comes back to the post page tab -> popup should show.

> AFAIK we rarely render the whole layout.

The `MainLayout` is rendered in almost every page, probably 96% of all pages (sample excluded ones are `onboarding`, referral page, etc.).

Reply:
What I meant by that comment is that when you click on the read post button, the main layout is not re-rendered, so the hook to show the modals would not be executed, increasing the delay that we show the modals.

I have moved the hook back to the `MainLayout` though, please take another look 🙏

Reply:
It would be re-rendered when you go back to the app (window focus) since the boot query will refetch, and it is inside the boot provider which will trigger the re-render and MainLayout is wrapped by the BootProvider.